File: kotlin/src/_704_binary_search.kt
/*
 * @lc app=leetcode id=704 lang=kotlin
 *
 * [704] Binary Search
 */

// @lc code=start
class Solution {
    fun search(nums: IntArray, target: Int): Int {
        var l = 0
        var r = nums.size - 1
        while (l <= r) {
            val mid = l + (r - l) / 2
            when {
                nums[mid] == target -> return mid
                nums[mid] < target -> l = mid + 1
                else -> r = mid - 1
            }
        }
        return -1
    }
}
// @lc code=end

File: typescript/src/_704_binary_search.ts
/*
 * @lc app=leetcode id=704 lang=typescript
 *
 * [704] Binary Search
 */

// @lc code=start
function search(nums: number[], target: number): number {
    let lo = 0, hi = nums.length - 1;
    while (lo <= hi) {
        const mid = lo + Math.floor((hi - lo) / 2);
        if (nums[mid] === target) return mid;
        else if (nums[mid] < target) lo = mid + 1;
        else hi = mid - 1;
    }
    return -1;
};
// @lc code=end
